lib/util/getLayersOfPcbElement.ts

```diff
@@ -0,0 +1,23 @@
+import type { Collidable } from "lib/check-each-pcb-trace-non-overlapping/getCollidableBounds"
+
+export function getLayersOfPcbElement(obj: Collidable): string[] {
+  if (obj.type === "pcb_trace_segment") {
+    return [obj.layer]
+  }
+  if (obj.type === "pcb_smtpad") {
+    return [obj.layer]
+  }
+  if (obj.type === "pcb_plated_hole") {
+    return Array.isArray(obj.layers) ? obj.layers : ["top", "bottom"]
+  }
+  if (obj.type === "pcb_hole") {
+    return ["top", "bottom"]
```

Comment:
have a var called `ALL_LAYERS` (or, even better, import from `circuit-json`) there are more than two layers for 4 and 8 layer boards etc.